src/main/rx/sumd.c

```diff
@@ -54,49 +54,57 @@
 #define SUMD_BUFFSIZE (SUMD_MAX_CHANNEL * 2 + 5) // 6 channels + 5 = 17 bytes for 6 channels
 
 #define SUMD_BAUDRATE 115200
+#define SUMD_TIME_NEEDED_PER_FRAME 4000
 
 static bool sumdFrameDone = false;
 static uint16_t sumdChannels[MAX_SUPPORTED_RC_CHANNEL_COUNT];
 static uint16_t crc;
 
 static uint8_t sumd[SUMD_BUFFSIZE] = { 0, };
 static uint8_t sumdChannelCount;
+static timeDelta_t lastFrameDelta = 0;
 
 // Receive ISR callback
 static void sumdDataReceive(uint16_t c, void *data)
 {
     UNUSED(data);
 
-    uint32_t sumdTime;
-    static uint32_t sumdTimeLast;
+    static timeUs_t sumdTimeLast;
     static uint8_t sumdIndex;
+    static timeUs_t lastFrameCompleteTimeUs = 0;
 
-    sumdTime = micros();
-    if ((sumdTime - sumdTimeLast) > 4000)
+    const timeUs_t sumdTime = micros();
+    if (cmpTimeUs(sumdTime, sumdTimeLast) > SUMD_TIME_NEEDED_PER_FRAME) {
         sumdIndex = 0;
+    }
     sumdTimeLast = sumdTime;
 
     if (sumdIndex == 0) {
-        if (c != SUMD_SYNCBYTE)
+        if (c != SUMD_SYNCBYTE) {
             return;
-        else
-        {
+        } else {
             sumdFrameDone = false; // lazy main loop didnt fetch the stuff
             crc = 0;
         }
     }
-    if (sumdIndex == 2)
+
+    if (sumdIndex == 2) {
         sumdChannelCount = (uint8_t)c;
-    if (sumdIndex < SUMD_BUFFSIZE)
+    }
+
+    if (sumdIndex < SUMD_BUFFSIZE) {
         sumd[sumdIndex] = (uint8_t)c;
+    }
+
     sumdIndex++;
-    if (sumdIndex < sumdChannelCount * 2 + 4)
+    if (sumdIndex < sumdChannelCount * 2 + 4) {
         crc = crc16_ccitt(crc, (uint8_t)c);
-    else
-        if (sumdIndex == sumdChannelCount * 2 + 5) {
-            sumdIndex = 0;
-            sumdFrameDone = true;
-        }
+    } else if (sumdIndex == sumdChannelCount * 2 + 5) {
```

Comment:
Would it make sense to translate these magic numbers into defines?

Reply:
Removed magic numbers and cleaned up the definition of the different parts of the protocol definition.